CDS/js/Datastore.js

```diff
@@ -114,24 +116,100 @@ function get(conn,type, key, callback, userdata) { /* callback as function(err,
                         console.error(err);
                         reject(err);
                     }
+                    var rtn;
                     if (!row) {
-                        fulfill({value: "(value not found)",type: type,key: key});
+                        rtn = {value: "(value not found)",type: type,key: key};
                     } else {
-                        if(callback){
-                            fulfill(callback(row.value,type,key,userdata));
-                        } else {
-                            fulfill({value: row.value,type: type,key: key});
-                        }
+                        rtn = {value: row.value,type: type,key: key}
+                    }
+
+                    if(callback){
+                        fulfill(callback(rtn.value, rtn.type, rtn.key, userdata));
+                    } else {
+                        fulfill(rtn);
                     }
                 });
             });
         });
     });
 }
 
+function substituteString(conn,str){
+    return new Promise(function(fulfill,reject) {
+
+        function replaceAllOccurances(string, matchText, replacementValue) {
+            //This replaces all occurances of matchText (by splitting) then replaces then with the value (by joining)
+            const sanitisedReplacement = replacementValue ? replacementValue : "undefined";
+            return string.split(matchText).join(sanitisedReplacement);
```

Comment:
possibly not in the scope of this PR as we've copy+pasted this block from below, but I'm a little scared of this, especially the `undefined` part. Why do we do this?